vapi/esx/settings/clusters/vms/transition.go

```diff
@@ -0,0 +1,226 @@
+}
+
+// VmSelectionType defines the different selection types for VM selection.
+type VmSelectionType string
+
+const (
+	// VmSelectionTypeVmExtraConfig selects System VMs that have a specific property configured in the VM
+	// extra configuration. The property has a
+	// key='com.vmware.vim.eam.selection'. The value represents a unique
+	// identifier used for VM selection and is provided by the client.
+	VmSelectionTypeVmExtraConfig VmSelectionType = "VM_EXTRA_CONFIG"
+)
+
+// VmSelectionSpec structure contains fields
+// to describe the criteria used to select System VMs to which an
+// AlternativeVmSpec configuration is applied.
+type VmSelectionSpec struct {
+	// SelectionType is the type for this VmSelectionSpec.
+	SelectionType VmSelectionType `json:"selection_type"`
+
+	// ExtraConfigValue is the unique VM extra configuration property value. The recommended usage is
+	// with an UUID.
+	//
+	// See VmSelectionTypeVmExtraConfig for more details.
+	ExtraConfigValue string `json:"extra_config_value"`
+}
+
+// TransitionSpec contains fields that describe the specification for transitioning a System VM Solution.
+type TransitionSpec struct {
+	// SourceCluster is the cluster to transition from.
+	SourceCluster types.ManagedObjectReference `json:"source_cluster"`
+
+	// Solution is the target desired solution specification in vLCM.
+	Solution *SolutionSpec `json:"solution"`
+}
+
+// EnableSpec contains fields that describe specification for enablement of EAM managed solution in vLCM.
+type EnableSpec struct {
+	// EamAgencyID is the identifier of the solution in EAM (EAM agency).
+	EamAgencyID string `json:"eam_agency_id"`
+
+	// Solution is the target desired solution specification in vLCM.
+	Solution *SolutionSpec `json:"solution"`
+}
+
+// MultiSourceEnableSpec contains fields that describe specification for enablement of multiple EAM managed solutions into single vLCM managed solution.
+// Supported only for solutions with deployment type DeploymentType#CLUSTER_VM_SET.
+type MultiSourceEnableSpec struct {
+	// EamAgencyIDs is the EAM Agency identifiers.
+	EamAgencyIDs []string `json:"eam_agency_ids"`
+
+	// Solution is the target desired solution specification in vLCM.
+	// The given SolutionSpec should not contain any AlternativeVmSpecs. See SourceAlternativeVmSpecs
+	// about how to configure AlternativeVmSpecs.
+	Solution *SolutionSpec `json:"solution"`
+
+	// SourceVmSelectionSpecs is the relation between System VMs and their respective VmSelectionSpecs.
+	// Provided VM IDs must be part of the solution being transitioned and
+	// must exist in the cluster where the solution is installed.
+	// Provided VmSelectionSpecs must be present in the applied desired state as part of the
+	// ClusterSolutionSpec#alternativeVmSpecs.
+	//
+	// If unset, no VmSelectionSpecs are applied on the source agencies' System VMs during the enablement.
+	SourceVmSelectionSpecs map[string]*VmSelectionSpec `json:"source_vm_selection_specs,omitempty"`
```

Comment:
Why is it a map of pointers? Are the `VmSelectionSpec` values somewhere else in this structure? I do not see them. Make this `map[string]VmSelectionSpec`.